src/util/pointProjection.ts

```diff
@@ -285,25 +280,30 @@ const matrixFromScaleOffset = (scale: scaleOffset, xScaleFactor: number, yScaleF
  * @returns A 2d scaling matrix which maps native data points into a desired drawing space.
  */
 export const use2dScalingMatrix = (props: ScalingProps) => {
-    // NOTE scaling may make us draw in the margins!!!! Think about this
     const {positiveYGrowsDownward, preserveAspectRatio} = props
-    const invert = !positiveYGrowsDownward
+    const {totalPixelWidth, totalPixelHeight, pixelMargins} = props
     const {dataXMin, dataXMax, dataYMin, dataYMax, yScaleFactor, xScaleFactor} = props
-    if (dataXMin === undefined || dataXMax === undefined || dataYMin === undefined || dataYMax === undefined) {
-        console.warn("Attempt to compute pixel projection matrix with undefined data elements. Mapping to null.")
-        return matrix([[0, 0, 0], [0, 0, 0]])
-    }
-    if (dataXMin === dataXMax && dataYMin === dataYMax) {
-        console.warn("Attempt to compute pixel projection matrix, but data has 0 width in both dimensions. Mapping to null.")
-        return matrix([[0, 0, 0], [0, 0, 0]])
-    }
-    
-    const _pixelDims = applyPixelMargins(props)
-    const dataDims = { xMin: dataXMin, xMax: dataXMax, yMin: dataYMin, yMax: dataYMax }
-    const scale = computeScales(_pixelDims, dataDims, invert, (preserveAspectRatio ?? false))
-    
-    const xform = matrixFromScaleOffset(scale, (xScaleFactor ?? 1), ((yScaleFactor ?? 1) * (invert ? -1 : 1)) )
-    return xform
+    const xformMatrix = useMemo(() => {
```

Comment:
Breaking out all the primitive data from `props` which the memoization call depends on--I don't want a situation where we have object-reference discrepancies breaking this memoization.